packages/block-library/src/categories/index.php

```diff
@@ -99,20 +99,26 @@ function render_block_core_categories( $attributes, $content, $block ) {
  */
 function build_dropdown_script_block_core_categories( $dropdown_id ) {
 	ob_start();
+
+	$exports = array( $dropdown_id, home_url() );
 	?>
 	<script>
-	( function() {
-		var dropdown = document.getElementById( '<?php echo esc_js( $dropdown_id ); ?>' );
-		function onCatChange() {
-			if ( dropdown.options[ dropdown.selectedIndex ].value !== -1 ) {
-				location.href = "<?php echo esc_url( home_url() ); ?>/?" + dropdown.name + '=' + dropdown.options[ dropdown.selectedIndex ].value;
+	( ( [ dropdownId, homeUrl ] ) => {
+		document.getElementById( dropdownId ).addEventListener( 'change', ( event ) => {
```

Comment:
Would it make sense to test the existence of incoming args? I get the possibility is remote that `$id` and `home_url()` won't render, but just in case DOM selector returns null for whatever reason

Reply:
Yeah, very remote possibility given that `$dropdown_id` is used right before to generate the dropdown. If an element with that ID doesn't exist in the DOM then probably the best outcome would be an error actually. So I don't think we need to do anything here, at least no such error handling was in there before.